api/src/models/request_info.rs

```diff
@@ -67,16 +45,32 @@ impl<'r> FromRequest<'r> for RequestInfo {
     type Error = ();
 
     async fn from_request(request: &'r Request<'_>) -> request::Outcome<Self, Self::Error> {
-        let hosts: Vec<_> = request.headers().get("host").collect();
+        let forwarded_host = request.headers().get_one("x-forwarded-host");
```

Comment:
This part of the code has been already merged in your previous PR #137. I think you have to rebase…